server/src/main/java/org/elasticsearch/rest/RestRequestFilter.java

```diff
@@ -63,7 +64,7 @@ public BytesReference content() {
                             throw new ElasticsearchException("failed to parse request", e);
                         }
                     }
-                    return filteredBytes;
+                    return ReleasableBytesReference.wrap(filteredBytes);
```

Comment:
Should we check the refcount of the original request before returning here?

Reply:
I do a copy above in request.requiredContent(), so it's GC backed reference without need for ref counts.

Reply:
This filter is tricky, it deserialize, filter, and serialize again. We definitely dont need original content anymore after it's called.